website/events/services.py

```diff
@@ -82,20 +82,20 @@ def event_permissions(member, event, name=None):
 
     perms["create_registration"] = (
         (registration is None or registration.date_cancelled is not None)
-        and (event.registration_allowed or not event.registration_required)
+        and (event.registration_allowed or event.optional_registration_allowed)
```

Comment:
I was thinking about this, and haven't tested.

But I think this allows registration for events that do require registration while that period hasn't opened yet.

Reply:
Pretty sure it should be

```suggestion
        and (event.registration_allowed or (event.optional_registration_allowed and not event.registration_required))
```

Reply:
maybe even move that logic into the model?

Reply:
you are right